bids/reports/utils.py

```diff
@@ -67,100 +65,52 @@ def list_to_str(lst):
     return str_
 
 
-def get_slice_info(slice_times):
+def clean_multiline(desc):
     """
-    Extract slice order from slice timing info.
-
-    TODO: Be more specific with slice orders.
-    Currently anything where there's some kind of skipping is interpreted as
-    interleaved of some kind.
-
-    Parameters
-    ----------
-    slice_times : array-like
-        A list of slice times in seconds or milliseconds or whatever.
-
-    Returns
-    -------
-    slice_order_name : :obj:`str`
-        The name of the slice order sequence.
-    """
-    # Slice order
-    slice_times = remove_duplicates(slice_times)
-    slice_order = sorted(range(len(slice_times)), key=lambda k: slice_times[k])
-    if slice_order == range(len(slice_order)):
-        slice_order_name = 'sequential ascending'
-    elif slice_order == reversed(range(len(slice_order))):
-        slice_order_name = 'sequential descending'
-    elif slice_order[0] < slice_order[1]:
-        # We're allowing some wiggle room on interleaved.
-        slice_order_name = 'interleaved ascending'
-    elif slice_order[0] > slice_order[1]:
-        slice_order_name = 'interleaved descending'
-    else:
-        slice_order = [str(s) for s in slice_order]
-        raise Exception('Unknown slice order: [{0}]'.format(', '.join(slice_order)))
-
-    return slice_order_name
-
-
-def get_seqstr(config, metadata):
+    Remove newlines and double-spaces from multiline string.
     """
-    Extract and reformat imaging sequence(s) and variant(s) into pretty
-    strings.
+    desc = desc.replace('\n', ' ').lstrip()
+    while '  ' in desc:
+        desc = desc.replace('  ', ' ')
+    return desc
 
-    Parameters
-    ----------
-    config : :obj:`dict`
-        A dictionary with relevant information regarding sequences, sequence
-        variants, phase encoding directions, and task names.
-    metadata : :obj:`dict`
-        The metadata for the scan.
 
-    Returns
-    -------
-    seqs : :obj:`str`
-        Sequence names.
-    variants : :obj:`str`
-        Sequence variant names.
-    """
-    seq_abbrs = metadata.get('ScanningSequence', '').split('_')
-    seqs = [config['seq'].get(seq, seq) for seq in seq_abbrs]
-    variants = [config['seqvar'].get(var, var) for var in \
-                metadata.get('SequenceVariant', '').split('_')]
-    seqs = list_to_str(seqs)
-    if seq_abbrs[0]:
-        seqs += ' ({0})'.format(os.path.sep.join(seq_abbrs))
-    variants = list_to_str(variants)
-    return seqs, variants
-
-
-def get_sizestr(img):
-    """
-    Extract and reformat voxel size, matrix size, field of view, and number of
-    slices into pretty strings.
+def collect_associated_files(layout, files, extra_entities=()):
```

Comment:
@tyarkoni This function is meant to group together files from the same acquisition, like multi-echo, complex (magnitude + phase), uncombined, or field map data. I thought that this probably had wider uses than report-building. Should it be integrated into another module, or is there too much entity/suffix hard-coding to fit in there?

Reply:
Is the issue that `BIDSFile.get_associations()` doesn't return all the associations you want, or is this something different? If the former, we should probably just add the desired associations at indexing (or at least discuss that possibility). If you don't think it makes sense to think of these as file-level associations, then it seems fine to keep it here, at least for now.

Reply:
Actually, given that it's a standalone function, I think we could certainly put it in `bids.utils` at minimum.

Reply:
Yeah, get_associations generally only returns json files in my experience, so things like other echoes for multi-echo data are not found. I do think it makes sense to count them as file-level associations, and would much rather rely on get_associations for this over a reports-specific function.

Reply:
I can open a new issue specifically for get_associations if that would help.

Reply:
Yeah, let's do that. That said, the indexing code is a mess and the association indexing is particularly slow, so I'm hesitant to do this as a band-aid... I might leave it open until a refactoring of the indexing happens, which could be a while. So for now I suggest moving this into `utils`, and if and when the association indexing is beefed up, we can revisit whether we still need the existing function. Thanks!

Reply:
👍 Will do.